libflux/src/semantic/nodes.rs

```diff
@@ -1134,8 +1148,8 @@ pub struct BooleanLit {
     pub value: bool,
 }
 
-impl BooleanLit {
-    fn infer(&self, env: Environment, f: &mut Fresher) -> Result {
+impl Infer for BooleanLit {
+    fn infer(&mut self, env: Environment, _f: &mut Fresher) -> Result {
```

Comment:
I think we should just remove `f` from the signature since it's not used.

Reply:
I would also prefer that `self` not be specified as mutable here (in the case of literals) since literals will not be mutated.